src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -101,13 +271,60 @@ def algorithm(self, parameters=None):
             with CodeTimer("Rucio block resolution", logger=logging):
                 self.findRucioBlocks(uniqueActivePU, msPileupList)
 
-            # TODO in future tickets: find the json files in the spec/sandbox area and tweak them
-
+            # loop over existing pileup names and
+            # find the json files in the spec/sandbox area and tweak them
+            adjustJSONSpec(msPileupList)
         except Exception as ex:
             msg = f"Caught unexpected exception in WorkflowUpdater. Details:\n{str(ex)}"
             logging.exception(msg)
             raise WorkflowUpdaterException(msg) from None
 
+    def adjustJSONSpec(self, msPileupList):
+        """
+        Main logic of the algorithm:
+        - for every pileup record find out location of tarball
+        - get configuration files within tarball
+        - extract block information
+        - replace pileupconf.json files within tarball
+        :param msPileupList: input list of pileup records
+        :return: nothing, it perform checks and adjust in place JSON spec file(s)
+        """
+        # loop over existing pileup names and
+        # find the json files in the spec/sandbox area and tweak them
+        for rec in msPileupList:
+            # find out location of pileup tarball
+            files = findFiles(self.idir, rec['pileupName'])
+            if len(files) != 1:
+                self.logger.warning("For pilupe %s found %s", rec['pileupName'], files)
+                continue
+            # name of sandbox tar file
+            tfile = files[0]
+
+            # get list of pileupconf.json files within our tar ball file
+            pileupConfFiles = findJsonSandboxFiles(tfile)
+
+            # extract blocks from msPileupList structures
+            blocks = rec['blocks']
+#                 blocks = [b for d in msPileupList for b in d["blocks"]]
+
+            # construct jdict={"path-to-json1": jdoc1, "path-to-json2": jdoc2}
+            jdict = {}
+            for fname in pileupConfFiles:
+                jdoc = extractPileupconf(tfile, fname)
+                bdict = blockLocations(jdoc)
+#                     jsonBlockNames = bdict.keys()
+
+                # Check if there is actually any pileup changes
+                rdict = rucioBlockLocations(self.rucio, blocks)
+                if not checkChanges(rdict, bdict):
+                    continue
+
+                updateBlockInfo(jdoc, rdict, bdict)
```

Comment:
We should have a log record before this update.

Reply:
ok, done